### DISCUSSION TITLE
Thoughts about employment overlay

### DISCUSSION THREAD

**Discussion by Ouaz** [Keriew/augustus#547] [Ideas]
Based on Crudelios idea: https://github.com/Keriew/augustus/discussions/544#discussioncomment-1756690

Some thoughts to possibly use as a start (a picture is worth a thousand words):
![employment_overlay_mockup](https://user-images.githubusercontent.com/11946570/145109945-68b91d48-e7b9-485e-ba36-5a63fcaa8363.png)

**Comment by crudelios** [comment]
I'd do it the other way around: the higher/ greener the column, the more employees the building had.

I audio wouldn't leave it empty for now employees, rather with the smallest column. Empty is for buildings that don't need employees.

**Reply by Ouaz** [reply]
Well, at first I hesitated, because I thought there will be green column everywhere (as no labor / reduced labor buildings are the exception most of the time). Anyway, your idea works very well:

![employment_overlay2](https://user-images.githubusercontent.com/11946570/145273486-673a437c-b67c-4099-9b90-9f860c04219e.png)

1, 2 and 3 require the same number of employees:
- 1 is fully staffed (let's say 10/10)
- 2 is understaffed (5/10) (50% height)
- 3 has no staff (0/10)

1 can have different heights based on the required total number of employees (2, 5, 8, 10, 20 etc).

As soon as a building is understaffed, the column turns to yellow (from 9/10 to 5/10, with its height corresponding from 90% to 50% of the full labor column height ), then to orange between 4/10 and 1/10 (height from 40% to 10%).

3 is always like that, for a building having no staff (no height).

**Reply by crudelios** [reply]
Hummm, won't different heights when there is full employment be confusing? Even with the color changes, there may be the case where a building that requires more employees and is lacking some has a larger column than one the requires fewer but has full employment.

**Reply by Ouaz** [reply]
![employment_overlay_height](https://user-images.githubusercontent.com/11946570/145286155-a6ba7b38-70b1-4215-b473-466de22969b1.png)

1 is full employment 5/5 (arbitrary height, I have no idea of the scale :p)
2 is partial employment 8/10

I don't know. Let's imagine a big city with plenty of full employment columns; yellow, orange, and red columns would be glaring among them anyway.
It would be also weird to have the same column height for a full-labor fountain and a full-labor hippodrome.

Or maybe to no make columns proportionnal to each others, but having "levels": ( < 10 employees) (10-50 employees) (> 50 employees).

**Comment by catalin-duta** [comment]
Just an idea, maybe something like the desirability overlay? Where the height is given by the number of employees *working* and the color indicates the employment percentage?
So for a building with 20/20 workers it is green and the same height as a building with 20/40, orange in color.

**Comment by crudelios** [comment]
I always found the desirability overlay too confusing... in my opinion the main objective of the employment overlay shouldn't be to show the max number of workers,  but to display in a glance whether they have full employment or not.

**Reply by catalin-duta** [reply]
Then how about:
- no employees = just red column base
- some employees = increase column height based on number of working employees, some color
- full employment = add column top and different color

So full employment, at a glance, is indicated by whether the column has a top or not.
